packages/core/src/services/instance/instance.service.ts

```diff
@@ -47,6 +47,11 @@ export interface ICreateUnitOptions {
  * It also manages the focused univer instance.
  */
 export interface IUniverInstanceService {
+    /** A set of callbacks that will be called before a new unit is added. */
```

Comment:
Should never expose internal properties as mutable. Instead, provide methods to do that.